scripts/loganalyze.py

```
@@ -0,0 +1,397 @@
+            err_sec.lines.append(f"    [{ln.timestamp}] {ln.source}; {ln.message}{detail}")
+    if init_fails:
+        # Collapse repeats of the same source/module into a single summary.
+        groups = {}
+        for ln, who in init_fails:
+            groups.setdefault(who, []).append(ln)
+
+        err_sec.lines.append(
+            f"{len(init_fails)} init-failure line(s) across {len(groups)} source(s):"
+        )
+        for who, lns in groups.items():
+            lns.sort(key=lambda l: l.timestamp)
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)